rsocket-routing-http-bridge/src/main/java/io/rsocket/routing/http/bridge/core/AbstractHttpRSocketFunction.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * @author Olga Maciaszek-Sharma
+ * @since 0.3.0
+ */
+abstract class AbstractHttpRSocketFunction<I, O> implements Function<I, O> {
+
+	public static final String TRANSPORT_KEY = "transport";
+	public static final String PORT_KEY = "port";
+	public static final String HOST_KEY = "host";
+	public static final String MAPPING_PATH_KEY = "mapping-path";
+
+	protected final Log LOG = getLog(getClass());
+
+	protected RoutingRSocketRequester defaultRequester;
+	protected RoutingRSocketRequesterBuilder requesterBuilder;
+	ObjectProvider<ClientTransportFactory> transportFactories;
+	protected final RSocketHttpBridgeProperties properties;
+	protected final Duration timeout;
+
+	protected AbstractHttpRSocketFunction(RoutingRSocketRequesterBuilder requesterBuilder,
+			RoutingRSocketRequester defaultRequester, ObjectProvider<ClientTransportFactory> transportFactories,
+			RSocketHttpBridgeProperties properties) {
+		this.requesterBuilder = requesterBuilder;
+		this.properties = properties;
+		timeout = properties.getTimeout();
+		this.transportFactories = transportFactories;
+		this.defaultRequester = defaultRequester;
+	}
+
+	protected void logTimeout(String address, String route) {
+		if (LOG.isDebugEnabled()) {
+			LOG.debug(String
+					.format("Timeout occurred while retrieving RSocket response from address: %s, route: %s. Response was not retrieved within %s", address, route, timeout));
+		}
+	}
+
+	protected void logException(Throwable error, String address, String route) {
+		if (LOG.isErrorEnabled())
+			LOG.error(String
+					.format("Exception occurred while retrieving RSocket response from address: %s, route: %s", address, route), error);
+	}
+
+	protected RoutingRSocketRequester getRequester(String brokerHeader) {
+		if (brokerHeader != null) {
+			TransportProperties broker = buildBroker(brokerHeader);
+			ClientTransport clientTransport = transportFactories
+					.orderedStream()
+					.filter(factory -> factory.supports(broker))
+					.findFirst()
+					.map(factory -> factory.create(broker))
+					.orElseThrow(() -> new IllegalStateException("Unknown transport: " + broker));
+			RoutingRSocketRequester requesterForBroker = requesterBuilder
+					.transport(clientTransport);
+			// if we don't subscribe, there won't be a connection to the broker.
+			requesterForBroker.rsocketClient().source().subscribe();
+			return requesterForBroker;
+		}
+		return defaultRequester;
+	}
+
+	protected TransportProperties buildBroker(String brokerData) {
+		String[] brokerDataPair = brokerData.toLowerCase().split(",");
+		Map<String, String> brokerDataMap = new HashMap<>();
+		Arrays.stream(brokerDataPair)
+				.map(pair -> Arrays.asList(pair.split("=")))
+				.filter(pair -> pair.size() == 2)
+				.forEach(pair -> brokerDataMap.put(pair.get(0), pair.get(1)));
+		String transport = brokerDataMap.get(TRANSPORT_KEY);
+		TransportProperties broker = new TransportProperties();
+		if (WEBSOCKET.name().equalsIgnoreCase(transport)) {
```

Comment:
Let's make host and port configurable via properties. If it's not in the header, use the properties. That way, in the case of being connected to a broker, client doesn't need to worry about it.

Reply:
They actually are :) I use `rsocket-routing-client-spring` underneath, so if the default (autowired) broker is used, it will be the one created with whatever routing-client setup the user has defined.

Reply:
where does that happen?

Reply:
For example [here](https://github.com/rsocket-routing/rsocket-routing-broker/pull/24/files#diff-823f23086bf2870311d7ed01f1cd947ab96fff2fb99dc7f1a437865731483ab1R97). It is [autowired](https://github.com/rsocket-routing/rsocket-routing-broker/pull/24/files#diff-027acdcc9835a216d7e5f85873fc7e003514c74978d22e42dd86f89611b19086R63).

Reply:
Actually, I think getting host information from a URL is a security issue. We'd either need an allow list or not allow it at all.

Reply:
It would be from header, but probably it's same thing. Maybe a whitelist would be ok?

Reply:
my initial reaction is to not have the option at all.

Reply:
Ok. Will remove it.